packages/@react-stately/data/src/useTreeData.ts

```diff
@@ -502,7 +502,8 @@ function moveItems<T extends object>(
       newMap = nextMap;
     }
     // decrement the index if the child being removed is in the target parent and before the target index
-    if (child.parentKey === toParent?.key
+    // eslint-disable-next-line eqeqeq
+    if (child.parentKey == toParent?.key
```

Comment:
```suggestion
    // the root node is special, it is null, and will not have a key, however, a parentKey can still point to it
    if ((child.parentKey === toParent
       || child.parentKey === toParent?.key)
```

I think this is a little more explicit as to what is being checked and doesn't need to ignore eslint.